src/services/refactors/inlineFunction.ts

```diff
@@ -0,0 +1,389 @@
+        const { program, file, startPosition } = context;
+        const info = getInfo(program, file, startPosition);
+        if (!info) return emptyArray;
+        const { allAvailable, selectedAvailable } = info;
+        const refactorInfo: ApplicableRefactorInfo = {
+            name: refactorName,
+            description: refactorDescription,
+            actions: []
+        };
+        if (allAvailable) {
+            refactorInfo.actions.push({
+                name: inlineAllActionName,
+                description: inlineAllActionDescription
+            });
+        }
+        if (selectedAvailable) {
+            refactorInfo.actions.push({
+                name: inlineHereActionName,
+                description: inlineHereActionDescription
+            });
+        }
+        return [refactorInfo];
+    }
+
+    function areSymbolsAccessible(checker: TypeChecker, symbols: ReadonlyArray<Symbol>, target: Node) {
+        return every(symbols, symbol => {
+            const symbolAccessibility = checker.isSymbolAccessible(
+                symbol,
+                target,
+                SymbolFlags.All,
+                /* shouldComputeAliasesToMakeVisible */ false).accessibility;
+            return symbolAccessibility === SymbolAccessibility.Accessible;
+        });
+    }
+
+    function getExternalSymbolsReferencedInScope(declaration: InlineableFunction, checker: TypeChecker) {
+        const ids = findDescendants(declaration, isIdentifier);
+        const visited = createMap<Symbol>();
+        forEach(ids, id => {
+            const symbol = checker.getSymbolAtLocation(id);
+            if (!symbol) return;
+            const symbolId = String(getSymbolId(symbol));
+            if (!visited.has(symbolId)) visited.set(symbolId, symbol);
+        });
+        const symbols = arrayFrom(visited.values());
+        return filter(symbols, s => {
+            if (s.valueDeclaration) {
+                const symbolScope = getEnclosingBlockScopeContainer(s.valueDeclaration);
+                return !(declaration === symbolScope || isAncestor(symbolScope, declaration));
+            }
+            return false;
+        });
+    }
+
+    function getInfo(program: Program, file: SourceFile, startPosition: number): Info | undefined {
+        const token = getTokenAtPosition(file, startPosition);
+        const checker = program.getTypeChecker();
+        if (isNameOfFunctionDeclaration(token)) {
+            return createInfo(checker, <InlineableFunction>token.parent);
+        }
+
+        const call = findAncestor(token, isCallExpression);
+        if (!call) return undefined;
+        const symbol = checker.getSymbolAtLocation(call.expression);
+        if (!symbol) return undefined;
+        const declaration = symbol.valueDeclaration;
+        if (!isInlineableFunction(declaration)) return undefined;
+        return createInfo(checker, declaration, call);
+    }
+
+    function createInfo(
+        checker: TypeChecker,

```

Comment:
If the parameter declaration is readable on single line, I would prefer single line over multiline
(same thing for following multiline parameter declaration)
like
```ts
function createInfo(checker: TypeChecker, declaration: InlineableFunction, call?: CallExpression): Info | undefined 
```

Reply:
Define "single line". How many characters?
I set max line length to 120. I don't like too long lines.